ark-core/src/proof_of_funds.rs

```diff
@@ -395,9 +396,16 @@ impl IntentMessage {
             .map_err(Error::ad_hoc)
             .context("failed to serialize intent message to JSON")
     }
+
+    pub fn to_delete(self) -> Self {
+        Self {
+            intent_message_type: IntentMessageType::Delete,
+            ..self
+        }
+    }
```

Comment:
We should undo these changes and tackle #79 properly, because did naive approach did not work.